wagtail/admin/models.py

```diff
@@ -68,3 +71,24 @@ def popular_tags_for_model(model, count=10):
         .annotate(item_count=Count("taggit_taggeditem_items"))
         .order_by("-item_count")[:count]
     )
+
+
+class EditingSession(models.Model):
+    user = models.ForeignKey(
+        get_user_model(),
+        on_delete=models.CASCADE,
+        related_name="editing_sessions",
+    )
+    content_type = models.ForeignKey(
+        ContentType, on_delete=models.CASCADE, related_name="+"
+    )
+    object_id = models.CharField(max_length=255)
+    content_object = GenericForeignKey(
+        "content_type", "object_id", for_concrete_model=False
+    )
```

Comment:
We'll likely want `RevisionMixin` to have a default `GenericRelation` to this model so that the object's `EditingSession`s are also cleaned up when the object is deleted (akin to #11750 which I haven't finished...)

Doesn't have to be in this same PR, can be a bit later when we've pieced everything together 😃 

Although, since we're only storing one content type here (which, by the way, would this be the most specific or the most basic type?), I wonder how smart would Django be when handling the deletion. I'd hope it won't matter because when you delete a Page, it would always delete both the row in the base table and the row in the specific one, so either of those should hopefully trigger the cleanup for the `GenericRelation`. But I haven't verified this, though!

Reply:
> would this be the most specific or the most basic type?

It's intended to be the base type, since "the wagtailcore.page with ID 123" is enough to uniquely identify the object. I wondered about ways to enforce that within the EditingSession model - either a check during `save`, or perhaps a custom 'create' helper method where you just have to pass the object and user - but eventually decided to postpone that detail until we had some more of the calling code in place...